From 5622b341ee98924fc44afdb5f3a3a115978e00fe Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sat, 18 Jul 2020 10:54:26 +0200
Subject: [PATCH 1/1] efi_loader: returning form UEFI FIT images

Do not reset the board when returning from an UEFI FIT image.

For failed UEFI binary we already print the return status in efi_run_image.
Remove duplicate output.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 common/bootm_os.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/common/bootm_os.c b/common/bootm_os.c
index 55296483f7..8b4397693a 100644
--- a/common/bootm_os.c
+++ b/common/bootm_os.c
@@ -542,15 +542,14 @@ static int do_bootm_efi(int flag, int argc, char *const argv[],
 	       images->ep);
 	bootstage_mark(BOOTSTAGE_ID_RUN_OS);
 
+	/* We expect to return */
+	images->os.type = IH_TYPE_STANDALONE;
+
 	image_buf = map_sysmem(images->ep, images->os.image_len);
 
 	efi_ret = efi_run_image(image_buf, images->os.image_len);
-	if (efi_ret != EFI_SUCCESS) {
-		printf("## Failed to run EFI image: r = %lu\n",
-		       efi_ret & ~EFI_ERROR_MASK);
+	if (efi_ret != EFI_SUCCESS)
 		return 1;
-	}
-
 	return 0;
 }
 #endif
-- 
2.27.0

